### PR TITLE
sync of pr/1824 changes -> main

### PR DESCRIPTION
**Pull request by malachi-constant**
### Relates 
#1571 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: awswrangler/dynamodb/_read.py [hunk 1]
```diff
@@ -272,9 +196,8 @@ def _read_items(
     return items
 
 
-def read_items(  # pylint: disable=too-many-branches
+def read_items(
     table_name: str,
-    index_name: Optional[str] = None,
```
Comment (LeonLuttenberger) [inline awswrangler/dynamodb/_read.py:277]: This change was made 4 hours ago, I don't think we need to remove it :)
https://github.com/aws/aws-sdk-pandas/pull/1961

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 38df65f036265955151d7b644e4565c9500ef636
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a9545a25-ba7f-4778-9688-3036fec2c715%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 2a097df87323b832037fe9123a189fa3e9593339
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8c4f806a-741b-4621-891f-405b4021d88f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
Closing in favor of #1964

**Pull request closed without merging**